syntax = "proto3";

package user;

service UserService {
  rpc GetUser (UserRequest) returns (UserResponse);
  rpc ListUsers (ListUsersRequest) returns (ListUsersResponse);
  rpc GetUserData (UserRequest) returns (UserDataResponse);
}

message UserRequest {
  int32 user_id = 1;
}

message UserResponse {
  int32 id = 1;
  string name = 2;
  string email = 3;
}

message ListUsersRequest {
  int32 limit = 1;
}

message ListUsersResponse {
  repeated UserResponse users = 1;
}

message UserDataResponse {
  int32 id = 1;
  string name = 2;
  string email = 3;
  string api_key = 4;
  string secret = 5;
}

